packages/react-accordion/src/components/AccordionHeader/AccordionHeader.types.ts

```diff
@@ -36,7 +36,8 @@ export type AccordionHeaderSlots = {
    * Expand icon slot rendered before (or after) children content in heading
    */
   icon?: IntrinsicShorthandProps<'div'>;
-  children: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>>;
+  // TODO: children should never be a slot, this should be refactored so that behavior is correct.
+  children: ComponentSlotProps<React.ComponentType<React.HTMLAttributes<HTMLElement>>>;
```

Comment:
I'd recommend leaving this as-is (except maybe add the TODO comment and log an issue to track the fix). The `ComponentSlotProps` type is just making it more complicated and isn't adding any value.

Reply:
Why shouldn't `children` be a slot?

Reply:
`children` is the list of child element(s) of the `root` slot (or technically, of the primary slot). It's not a slot itself.

Reply:
Removed the change and just left the TODO comment.